CSharpMath.Forms/MathInputButton.cs

```diff
@@ -1,35 +1,72 @@
+    }
+    public static readonly BindableProperty PlaceholderActiveColorProperty = BindablePropertyWithButtonDraw<MathInputButton>(nameof(PlaceholderActiveColor), typeof(Color), defaultValue: NullableColorBindablePropertyHelper.Null);
+    public Color? PlaceholderRestingColor {
+      get => this.GetNullableColor(PlaceholderRestingColorProperty);
+      set => this.SetNullableColor(PlaceholderRestingColorProperty, value);
+    }
+    public static readonly BindableProperty PlaceholderRestingColorProperty = BindablePropertyWithButtonDraw<MathInputButton>(nameof(PlaceholderRestingColor), typeof(Color), defaultValue: NullableColorBindablePropertyHelper.Null);
+    public override void ButtonDraw() {
+      Content ??= new TextView();
+      switch (Input) {
+        case MathKeyboardInput.Left: Content.LaTeX = "\u25C0"; break;
+        case MathKeyboardInput.Right: Content.LaTeX = "\u25B6"; break;
+        case MathKeyboardInput.Up: Content.LaTeX = "\u25B2"; break;
+        case MathKeyboardInput.Down: Content.LaTeX = "\u25BC"; break;
+        case MathKeyboardInput.Backspace: Content.LaTeX = "\u2B05"; break;
+        case MathKeyboardInput.Return: Content.LaTeX = "\u21B5"; break;
+        case MathKeyboardInput.Clear: Content.LaTeX = "\u21BB"; break;
+        case MathKeyboardInput.Dismiss: Content.LaTeX = "\u2A2F"; break;
+        case MathKeyboardInput.Space: Content.LaTeX = @"\ ␣\ "; break;
         default:
           var keyboard = new MathKeyboard();
-          keyboard.KeyPress(input);
-          return keyboard.LaTeX;
+          keyboard.KeyPress(Input);
+          var latex = keyboard.LaTeX;
+          Color? restingPlaceholderColor = PlaceholderRestingColor ?? LaTeXSettings.PlaceholderRestingColor;
+          Color? activePlaceholderColor = PlaceholderActiveColor ?? LaTeXSettings.PlaceholderActiveColor;
+          if (restingPlaceholderColor != null || activePlaceholderColor != null) {
+            var restingNucleus = LaTeXSettings.PlaceholderRestingNucleus.Replace("\u25A1", @"\square");
+            var coloredPlaceholderRestingNucleus = LatexHelper.SetColor(restingNucleus, restingPlaceholderColor);
+            var coloredPlaceholderActiveNucleus = LatexHelper.SetColor(LaTeXSettings.PlaceholderActiveNucleus, activePlaceholderColor);
+            if (Input == MathKeyboardInput.Power || Input == MathKeyboardInput.Subscript) {
+              latex = latex.ReplaceFirstOccurrence(restingNucleus, coloredPlaceholderRestingNucleus);
+              latex = latex.ReplaceLastOccurrence(LaTeXSettings.PlaceholderActiveNucleus, coloredPlaceholderActiveNucleus);
+            } else if (LaTeXSettings.PlaceholderRestingNucleus == LaTeXSettings.PlaceholderActiveNucleus && latex.SubStringCount(LaTeXSettings.PlaceholderActiveNucleus) > 1
+                || restingNucleus != LaTeXSettings.PlaceholderActiveNucleus && latex.Contains(restingNucleus) && latex.Contains(LaTeXSettings.PlaceholderActiveNucleus)) {
```

Comment:
One minor thing: can you add a comment to this compound condition? The meaning is not totally obvious. Have feeling (probably a wrong one) that some brackets are missing.

Reply:
I totally agree that that piece of code is not very good readable. However, I'm not a fan of comments in code. Maybe I should refactor it? This could be commented code:
```
// unicode "\u25A1" is converted into latex "\square". Why only take this into account for the resting placeholder? I guess that the active placeholder will never be an empty square.
            var restingNucleus = LaTeXSettings.PlaceholderRestingNucleus.Replace("\u25A1", @"\square");
            var coloredPlaceholderRestingNucleus = LatexHelper.SetColor(restingNucleus, restingPlaceholderColor);
            var coloredPlaceholderActiveNucleus = LatexHelper.SetColor(LaTeXSettings.PlaceholderActiveNucleus, activePlaceholderColor);
            if (Input == MathKeyboardInput.Power || Input == MathKeyboardInput.Subscript) {
// In case of a power or subscript, the placeholder on the RIGHT is active on first appearance.
              latex = latex.ReplaceFirstOccurrence(restingNucleus, coloredPlaceholderRestingNucleus);
              latex = latex.ReplaceLastOccurrence(LaTeXSettings.PlaceholderActiveNucleus, coloredPlaceholderActiveNucleus);
            } else if (
// in ALL other cases of two placeholders,
// with the same nucleus (both a filled square, but with a different color)
LaTeXSettings.PlaceholderRestingNucleus == LaTeXSettings.PlaceholderActiveNucleus && latex.SubStringCount(LaTeXSettings.PlaceholderActiveNucleus) > 1
// or with a different nucleus, a filled square vs an empty square
                 || restingNucleus != LaTeXSettings.PlaceholderActiveNucleus && latex.Contains(restingNucleus) && latex.Contains(LaTeXSettings.PlaceholderActiveNucleus)) {
// the placeholder on the LEFT should be active on first appearance
              latex = latex.ReplaceFirstOccurrence(LaTeXSettings.PlaceholderActiveNucleus, coloredPlaceholderActiveNucleus);
              latex = latex.ReplaceLastOccurrence(restingNucleus, coloredPlaceholderRestingNucleus);
            } else {
// single placeholder should be active (let's not overwrite, but set the right color in one go)
              latex = latex.Replace(LaTeXSettings.PlaceholderActiveNucleus, coloredPlaceholderActiveNucleus);
            }
```
I realize that refactoring the code with same functional results, but wrong latex, is possible. I will add another unit test.

Reply:
Good enough for me. Yeah, I think refactoring makes sense but it can be done it later, not in this PR.